packages/core/src/lib/select/searchable-select.svelte

```diff
@@ -1,241 +1,261 @@
 <!--
 @component
 
-For selecting from a list of options.
+Select an option from a list, with search
 
 ```svelte
 <SearchableSelect
   options={["Option 1", "Option 2", "Option 3"]}
-  on:input={onSelect}
+  bind:value
 />
 ```
 -->
-<svelte:options immutable />
-
 <script lang="ts">
 import cx from 'classnames';
-import { uniqueId } from '$lib/unique-id';
-import { clickOutside } from '$lib/click-outside';
-import type { IconName } from '$lib/icon';
+import { Floating } from '$lib/floating';
+import { Icon } from '$lib/icon';
 import { InputStates, type InputState } from '$lib/input';
+import { createHandleKey } from '$lib/keyboard';
+import { uniqueId } from '$lib/unique-id';
 
-import { selectControls } from './controls';
-import { createSearchableSelectDispatcher } from './dispatcher';
 import { SortOptions, getSearchResults, type SortOption } from './search';
 import SelectInput from './select-input.svelte';
-import SelectMenuButton from './select-menu-button.svelte';
-import SelectMenu from './select-menu.svelte';
 
 /** The options the user should be allowed to search and select from. */
-export let options: string[] = [];
+export let options: string[];
 
 /** The value of the search input or the currently selected option, if any. */
-export let value: string | undefined = undefined;
+export let value = '';
 
-/** Whether or not the select should be rendered as disabled and be non-operable. */
+/** The placeholder of the input. */
+export let placeholder = '';
+
+/**
+ * Whether value must be constrained to `options` on change.
+ * - `false` (Default) - any value may be selected
+ * - `true` - value must be in `options`
+ * - `Function` - value must be in `options` or pass the test function
+ */
+export let exclusive: boolean | ((value: string) => boolean) = false;
```

Comment:
This is a change after using this component in the app; the data UI needs a little more involved filtering than "is the value in the options"